tlslite/utils/codec.py

```diff
@@ -3,31 +3,195 @@
 
 """Classes for reading/writing binary data (such as TLS records)."""
 
-from .compat import *
+import sys
+import struct
+from struct import pack
+
 
 class Writer(object):
+    """Serialisation helper for complex byte-based structures."""
+
     def __init__(self):
+        """Initialise the serializer with no data."""
         self.bytes = bytearray(0)
 
-    def add(self, x, length):
-        self.bytes += bytearray(length)
-        newIndex = len(self.bytes) - 1
-        for count in range(length):
-            self.bytes[newIndex] = x & 0xFF
-            x >>= 8
-            newIndex -= 1
-        if x != 0:
-            raise ValueError("Can't represent value in specified length")
+    def addOne(self, val):
+        """Add a single-byte wide element to buffer, see add()."""
+        self.bytes.append(val)
+
+    if sys.version_info < (2, 7):
+        # struct.pack on Python2.6 does not raise exception if the value
+        # is larger than can fit inside the specified size
+        def addTwo(self, val):
+            """Add a double-byte wide element to buffer, see add()."""
+            if not 0 <= val <= 0xffff:
+                raise ValueError("Can't represent value in specified length")
+            self.bytes += pack('>H', val)
+
+        def addThree(self, val):
+            """Add a thee-byte wide element to buffer, see add()."""
+            if not 0 <= val <= 0xffffff:
+                raise ValueError("Can't represent value in specified length")
+            self.bytes += pack('>BH', val >> 16, val & 0xffff)
+
+        def addFour(self, val):
+            """Add a four-byte wide element to buffer, see add()."""
+            if not 0 <= val <= 0xffffffff:
+                raise ValueError("Can't represent value in specified length")
+            self.bytes += pack('>I', val)
+    else:
+        def addTwo(self, val):
+            """Add a double-byte wide element to buffer, see add()."""
+            try:
+                self.bytes += pack('>H', val)
+            except struct.error:
+                raise ValueError("Can't represent value in specified length")
+
+        def addThree(self, val):
+            """Add a thee-byte wide element to buffer, see add()."""
```

Comment:
thanks!

since it's so minor, I'll just add new commit that fixes it instead of rebasing